src/Implement/ResponseHandler.cpp

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2021, 鍾淯丞, 周杰仕, 林仁鴻. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <iostream>
+#include <sys/socket.h>
+
+#include "OperateData.h"
+#include "ResponseHandler.h"
+#include "ResponseEncoder.h"
+
+void ResponseHandler::Send(const std::string &for_client_fd, const std::unique_ptr<OperateData> &data) {
+    
+    try {
+        ResponseEncoder *tmp_encoder = ResponseEncoder::GetInstance();
+        // Encode header and body to OperateData
+        std::string response = tmp_encoder->Encode(data);
```

Comment:
You might have to change ResponseEncoder::Encode() to take unique_ptr<> without the const and without the reference as well.

Then, you'll pass `std::move(data)` into the method to transfer the ownership. Assuming you're transferring the ownership of `data` in for it to be destroyed there.

Reply:
Done!